packages/blade/src/components/Radio/RadioIcon/RadioIcon.tsx

```diff
@@ -1,55 +1,55 @@
 /* eslint-disable @typescript-eslint/explicit-function-return-type */
 /* eslint-disable @typescript-eslint/explicit-module-boundary-types */
 import React from 'react';
+import { radioIconColors, radioSizes } from '../radioTokens';
+import type { RadioProps } from '../Radio';
 import { RadioIconWrapper } from './RadioIconWrapper';
 import { Fade } from './Fade';
 import { useTheme } from '~components/BladeProvider';
 import Svg from '~components/Icons/_Svg';
 import Circle from '~components/Icons/_Svg/Circle';
-import { getIn } from '~utils';
-
-const CheckedIcon = ({ color }: { color: string }) => {
-  return (
-    <Svg width="16px" height="16px" viewBox="0 0 16 16" fill="none">
-      <Circle cx="8" cy="8" r="4" fill={color} />
-    </Svg>
-  );
-};
+import { getIn, makeSpace } from '~utils';
 
 export type RadioIconProps = {
   isDisabled?: boolean;
   isNegative?: boolean;
   isChecked?: boolean;
-};
+} & Required<Pick<RadioProps, 'size'>>;
+
+const CheckedIcon = ({ color, size }: { color: string; size: RadioIconProps['size'] }) => {
+  const width = radioSizes.icon[size].width;
```

Comment:
This logic seems too complex imo. Easier to just have multiple sets of icons as different components for different sizes

Reply:
Lets do this instead:

https://github.com/razorpay/blade/pull/758#discussion_r1004392364

Reply:
Doing it this way because the `getRadioIconWrapperStyles`'s width & height also needs to match the radio's icon sizing. 

<img width="491" alt="Screenshot 2022-10-25 at 5 35 57 PM" src="https://user-images.githubusercontent.com/35374649/197768752-ef810638-d5a1-4523-8014-baf2dcc53263.png">

Thus colocated, it's easier to just make the change in one place instead of jumping files to find where I need to make the sizing changes.